src/_P036_FrameOLED.ino

```diff
@@ -969,46 +1192,58 @@ boolean Plugin_036(uint8_t function, struct EventStruct *event, String& string)
         }
         # endif // P036_ENABLE_LINECOUNT
         # ifdef P036_ENABLE_LEFT_ALIGN
-        else if ((subcommand == F("leftalign")) &&
+        else if ((subcommand == F("leftalign")) && // for compatibility
                  ((event->Par2 == 0) ||
                   (event->Par2 == 1))) {
           success = true;
-          P036_data->setTextAlignment(event->Par2 == 1 ? TEXT_ALIGN_LEFT : TEXT_ALIGN_CENTER);
-          bitWrite(P036_FLAGS_0, P036_FLAG_LEFT_ALIGNED, event->Par2); // Set in settings
+          eAlignment aAlignment = (event->Par2 == 1 ? eAlignment::eLeft : eAlignment::eCenter);
+          P036_data->setTextAlignment(aAlignment);
+          uint32_t lSettings = P036_FLAGS_1;
+          set2BitToUL(lSettings, P036_FLAG_LEFT_ALIGNED, static_cast<uint8_t>(aAlignment)); // Alignment
+          P036_FLAGS_1 = lSettings;
+        }
+        else if ((subcommand == F("align")) &&
+                 ((event->Par2 == 0) ||
+                  (event->Par2 == 1) ||
+                  (event->Par2 == 2))) {
+          success = true;
+          eAlignment aAlignment;
+
+          switch (event->Par2) {
+            case 1: aAlignment  = eAlignment::eLeft; break;
+            case 2: aAlignment  = eAlignment::eRight; break;
+            default: aAlignment = eAlignment::eCenter; break;
+          }
+          P036_data->setTextAlignment(aAlignment);
+          uint32_t lSettings = P036_FLAGS_1;
+          set2BitToUL(lSettings, P036_FLAG_LEFT_ALIGNED, static_cast<uint8_t>(aAlignment)); // Alignment
+          P036_FLAGS_1 = lSettings;
         }
         # endif // ifdef P036_ENABLE_LEFT_ALIGN
         else if ((LineNo > 0) &&
                  (LineNo <= P36_Nlines)) {
           // content functions
           success = true;
-          String NewContent = parseStringKeepCase(string, 3);
-          NewContent = P036_data->P36_parseTemplate(NewContent, 20);
-
-          if (!safe_strncpy(P036_data->DisplayLinesV1[LineNo - 1].Content, NewContent, P36_NcharsV1)) {
-            addHtmlError(getCustomTaskSettingsError(LineNo - 1));
-          }
-          P036_data->DisplayLinesV1[LineNo - 1].Content[P36_NcharsV1 - 1] = 0;                    // Terminate in case of uninitalized data
-          P036_data->DisplayLinesV1[LineNo - 1].reserved                  = (event->Par3 & 0xFF); // not implemented yet
+          String* currentLine = &P036_data->LineContent->DisplayLinesV1[LineNo - 1].Content;
+          *currentLine = parseStringKeepCase(string, 3);
+          *currentLine = P036_data->P36_parseTemplate(*currentLine, LineNo - 1);
```

Comment:
I don't get it why the minimal line length should be based on the line nr.

Reply:
The function `P36_parseTemplate()` has the `minimal_lineSize` of 20 hard coded inside.
The `lineIdx` (LineNo-1) is necessary to get the alignment for this line to handle the leading spaces for left aligned lines as requested by tonihuisman.